browser/ui/views/toolbar/brave_vpn_button.cc

```diff
@@ -221,12 +227,14 @@ std::unique_ptr<views::Border> BraveVPNButton::GetBorder(
 }
 
 void BraveVPNButton::UpdateColorsAndInsets() {
+  const bool is_connect_error = IsConnectError();
+  is_error_state_ = is_connect_error;
```

Comment:
I think `UpdateColorsAndInsets()` is not good place to set `is_error_state_`.
Could we do it in `OnConnectionStateChanged()`?

Reply:
`UpdateColorsAndInsets` is also called from `OnPurchasedStateChanged`

Reply:
How about having another method only for it and call it from both places?
As `UpdateColorsAndInsets()` is for updating UI based on current state,
updating error state from it seems not appropriate.

And it should be called when purchased state is changed?

Reply:
done